### PR TITLE
Watch later and playlist

### PR DESCRIPTION
**Pull request by omkarpatke**
# Added PlayList and WatchLater Functionality

**Screenshots:**
## WatchLater 
watchlater1
watchlater2
## PlayList

playlist
playlist2

**Files to check: **
- WatchLater.js
- WatchLater.css
- PlayList.js
- PlayList.css
- PlayListVideos.js
- SingleVideoPage.js
- api-calls.js
- PlayListModal.js
- PlayListContext.js
- watchlaterContext.js

### CODE REVIEW

**Comment by Shivani-1524** [comment]
functionality not working in preview please recheck

File: src/Pages/PlayListVideos/PlayListVideos.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    videos = playlists.find(item => item._id === playListVideosId).videos
+  }else{
+    videos = [];
+  }
+  return (
+    <div className='page-container'>
+    <SideBar />
+    <div className="home-section">
+    <div className='play-lists'>
+    {videos && videos.map(video => (
```
Comment (yaswanthmyneni) [inline src/Pages/PlayListVideos/PlayListVideos.js:27]: suggestion: can use the optional chaining operator here.

File: src/Pages/PlayListVideos/PlayListVideos.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <div key={video._id} className='video'>
+        <Link to={`/videos/${video._id}`}><img src={video.image} alt="videoImg" className='video-img like-video-img' /></Link>
+        <div className="dislike-btn" title='Delete Video' onClick={async() => {
+           const response = await removeVideoFromPlaylist( playListVideosId, video._id);
+           playListDispatch({type : 'REMOVE_DATA_FROM_PLAYLIST' , payload:response})
```
Comment (yaswanthmyneni) [inline src/Pages/PlayListVideos/PlayListVideos.js:32]: changes: check status and dispatch everywhere

File: src/context/PlayListContext.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+            case 'REMOVE_DATA_FROM_PLAYLIST':
+                return {...accu , type: action.type , payload : action.payload}
+    
+            default : 
+               return {...accu}
+        }
+    }
+
+    const [playListState , playListDispatch] = useReducer(playlistReducer , {type:'none' , payload:'none'})
```
Comment (yaswanthmyneni) [inline src/context/PlayListContext.js:34]: changes: i think type is not required in playListState

**Review by yaswanthmyneni — APPROVED** [review]
You can merge after changes

**Review by Shivani-1524 — CHANGES_REQUESTED** [review]
functionality not working in preview please recheck. also, as you scroll the videos appear on top of the navbar.

**Pull request merged**